### PR TITLE
[tf.keras] Enable tf.keras.callbacks.CallbackList API for tf.keras

### PR DESCRIPTION
**Pull request by ziky90**
In keras I'm used to use `keras.callbacks.CallbackList` I'd like to be able to use it also in the tf.keras this PR should enable the API `tf.keras.callbacks.CallbackList`.

### CODE REVIEW

**Comment by ziky90** [comment]
@karmel @harshini-gadige any progress reviewing this PR? It'd be nice to have this in one of the next tensorflow releases. I assume that this might be beneficial for people that are used to standard keras in order to have the same API also for the `tf.keras`.

**Comment by karmel** [comment]
Thanks for the PR. Can you clarify why you want to use CallbackList as a public API?

**Comment by karmel** [comment]
Assigning to @fchollet for thoughts on exposing CallbackList to the public API.

**Comment by ziky90** [comment]
@fchollet @karmel any progress reviewing this PR.

**Comment by ziky90** [comment]
@fchollet @karmel @harshini-gadige any progress reviewing and possibly merging this one-line PR?

**Comment by harshini-gadige** [comment]
> @fchollet @karmel @harshini-gadige any progress reviewing and possibly merging this one-line PR?

@karmel  @fchollet  Any update please ?

**Comment by karmel** [comment]
@omalleyt12 do you see any downsides here? I am okay with this in theory.

**Comment by omalleyt12** [comment]
> @omalleyt12 do you see any downsides here? I am okay with this in theory.

Hm, I'm torn because It's mostly an implementation detail, but I see the benefit for the use @ziky90 is mentioning. I don't see `CallbackList` documented in external Keras docs https://keras.io/callbacks/ 

2 potential downsides are:

1) Users might do: `model.fit(x, y, callbacks=keras.callbacks.CallbackList([cb1, cb2]))` which will actually short-circuit our Callback configuration logic as it's currently written

2) If anyone subclasses this class and overrides methods for use in Keras's own fit/eval/predict, there will be some unintuitive behavior. (Methods like `on_test_batch_begin` are never called directly in these loops, instead `_call_batch_hook` is called). But I don't see many users wanting to do that.

@ziky90 you ought to be able to do:

```python
from tensorflow.python.keras.callbacks import CallbackList
```

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @karmel, @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by karmel** [comment]
@ziky90 -- it sounds like the direct import is working for you? Can we close this PR? Thanks--

**Comment by ziky90** [comment]
@karmel It works for me, so we can close the PR for now.

**Comment by QRabbani** [comment]
I'm not sure where to submit this, but this line of code wasn't in the 2.0 version I built from source. I added it in manually to my local version, but it still did not fully fix the problem.

Basically, I can use 
`from tensorflow.python.keras.callbacks import CallbackList`
but not
`from tensorflow.keras.callbacks import CallbackList`
after adding the line of code in the original pull request above.

To fix this, I had to go into the machine-generated code (not ideal I know) in "tensorflow/_api/v1/keras/callbacks/__init__.py" and add
`from tensorflow.python.keras.callbacks import CallbackList`

Perhaps someone could reopen this pull request, add that to the build code (that creates the machine-generated code) and close it? That's kind of outside of my knowledge. Thanks!
